config/src/config/secure_config.rs

```diff
@@ -66,7 +66,6 @@ pub struct VaultConfig {
 pub struct OnDiskStorageConfig {
     // Required path for on disk storage
     pub path: PathBuf,
-    #[serde(skip)]
```

Comment:
Was this an accident? This should not be configured here and will result in weird issues if set_data_dir is called by other functions. Most likely you need to add the appropriate handlers to other configs.

Reply:
Thanks for pointing this out! This was because set_data_dir() was not being called on secure config, which I've now fixed.